### PR TITLE
[DLS-8266] Adding check your answers sites

### PR DESCRIPTION
**Pull request by JakeReid2020**

### CODE REVIEW

File: conf/messages.en [hunk 1]
```diff
@@ -174,7 +175,7 @@ askSecondaryWarehouses.detailsLink = Why should I register a warehouse?
 askSecondaryWarehouses.detailsContent = You can delay the point you will have to report liable drinks in a return and pay the levy on them if you register the warehouses you use to store them.
 askSecondaryWarehouses.checkYourAnswersLabel = Do you want to register any of the UK warehouses you use to store liable drinks?
 askSecondaryWarehouses.error.required = Select yes if you want to register any of the UK warehouses you use to store liable drinks
-askSecondaryWarehouses.change.hidden = AskSecondaryWarehouses
+askSecondaryWarehouses.change.hidden = Ask Secondary Warehouses
```
Comment (Chrissystk) [inline conf/messages.en:178]: This hidden text comes following "Change" so "Change Ask Secondary Warehouses" does not make sense. It should be something like "the UK warehouses you use to store liable drinks"

File: conf/messages.en [hunk 2]
```diff
@@ -257,7 +258,7 @@ packAtBusinessAddress.title = Your business address
 packAtBusinessAddress.heading2 = Is your registered business address a UK packaging site that you operate to produce liable drinks?
 packAtBusinessAddress.checkYourAnswersLabel = Is your registered business address a UK packaging site that you operate to produce liable drinks?
 packAtBusinessAddress.error.required = Select yes if packAtBusinessAddress
-packAtBusinessAddress.change.hidden = PackAtBusinessAddress
+packAtBusinessAddress.change.hidden = Pack At Business Address
```
Comment (Chrissystk) [inline conf/messages.en:261]: Same as above here should be the UK packaging site that you operate to produce liable drinks.

**Review by Chrissystk — DISMISSED** [review]

File: app/views/summary/UKSitesSummary.scala [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+package views.summary
+
+import controllers.routes
+import models.{CheckMode, UserAnswers}
+import pages.{AskSecondaryWarehousesPage, PackAtBusinessAddressPage}
+import play.api.i18n.Messages
+import uk.gov.hmrc.govukfrontend.views.Aliases.SummaryList
+import viewmodels.govuk.summarylist._
+import viewmodels.implicits._
+
```
Comment (RaminEsfandiari) [inline app/views/summary/UKSitesSummary.scala:26]: This function can be refactored to be more clean by saving your two `SummaryListRowViewModel`  in appropriately named defs outside of your summaryList def. You can then call them in your match cases. 

Pseudo code would be something in the lines of:

```
def summaryList(x,y) = {

    (a,b) match {
         case () =>
             getPackAtBusinessAddressRow
         case () =>
             getAskSecondaryWarehouseRow
         case () => 
             getPackAtBusinessAddressRow
             getAskSecondaryWarehouseRow
    }

}

def getPackAtBusinessAddressRow(...)
def getAskSecondaryWarehouseRow(...)
```

File: conf/messages.en [hunk 4]
```diff
@@ -66,6 +66,7 @@ checkYourAnswers.businessDetails.utr = Unique Taxpayer Reference (UTR)
 checkYourAnswers.businessDetails.name = Business name
 checkYourAnswers.businessDetails.address = Business address
 checkYourAnswers.businessDetails.litresGlobally = Litres of your own brands of liable drinks packaged globally in the past 12 months
+checkYourAnswers.sites = Uk site details
```
Comment (RaminEsfandiari) [inline conf/messages.en:69]: capitalise k in Uk

File: app/views/summary/UKSitesSummary.scala [hunk 5]
```diff
@@ -0,0 +1,111 @@
+/*
```
Comment (RaminEsfandiari) [inline app/views/summary/UKSitesSummary.scala:1]: There are no tests to cover your first two and the last scenarios for this view. Please add those.

image

**Review by RaminEsfandiari — CHANGES_REQUESTED** [review]

File: app/views/summary/UKSitesSummary.scala (hunk 5 above)
Comment (JakeReid2020) [reply app/views/summary/UKSitesSummary.scala:1]: image

**Review by RaminEsfandiari — DISMISSED** [review]

**Comment by RaminEsfandiari** [comment]
@JakeReid2020 Approved but there is a compilation error on CheckYourAnswersControllerISpec. Let me know when resolved and I'll re-review

**Review by RaminEsfandiari — DISMISSED** [review]

**Review by ChrisHillCap — CHANGES_REQUESTED** [review]
waiting for convo

**Review by RaminEsfandiari — APPROVED** [review]

File: app/views/summary/UKSitesSummary.scala [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        key =  if(userAnswers.packagingSiteList.size > 1){
+          messages("packagingSiteDetails.checkYourAnswersLabel.multiple",  {userAnswers.packagingSiteList.size.toString})}else{
+          messages("packagingSiteDetails.checkYourAnswersLabel.one")
+        },
+        value = Value(),
+        actions = if (isCheckAnswers) {
+          Seq(
+            ActionItemViewModel("site.change", routes.PackAtBusinessAddressController.onPageLoad(CheckMode).url)
+              .withAttribute(("id", "change-packaging-sites"))
+              .withVisuallyHiddenText(messages("packAtBusinessAddress.change.hidden"))
```
Comment (ChrisHillCap) [inline app/views/summary/UKSitesSummary.scala:45]: needs amending

File: app/views/summary/UKSitesSummary.scala [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        key = if(userAnswers.warehouseList.size > 1){
+          messages("warehouseDetails.checkYourAnswersLabel.multiple",  {userAnswers.warehouseList.size.toString})}else{
+          messages("warehouseDetails.checkYourAnswersLabel.one")
+        },
+        value = Value(),
+        actions = if (isCheckAnswers) {
+          Seq(
+            ActionItemViewModel("site.change", routes.AskSecondaryWarehousesController.onPageLoad(CheckMode).url)
+              .withAttribute(("id", "change-warehouse-sites"))
+              .withVisuallyHiddenText(messages("askSecondaryWarehouses.change.hidden"))
```
Comment (ChrisHillCap) [inline app/views/summary/UKSitesSummary.scala:64]: needs amending

**Review by ChrisHillCap — APPROVED** [review]

**Pull request merged**